xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/migration/AbstractDataMigrationManager.java

```diff
@@ -366,6 +367,49 @@ public void initialize() throws InitializationException
         this.observationManager.addListener(new WikiDeletedEventListener());
     }
 
+    public boolean isMigrationIgnored(DataMigration migration, Set<String> ignoredMigrations)
+    {
+        return ignoredMigrations.contains(migration.getClass().getName())
+            || ignoredMigrations.contains(migration.getVersion().toString());
+    }
+
+    /**
+     * Ensure we don't have two migrations with same DB version.
+     *
+     * @throws InitializationException in case we found same DB version or the list of migration cannot be obtained.
+     */
+    private void checkMigrationsVersions() throws InitializationException
+    {
+        try {
+            List<HibernateDataMigration> migrationList =
+                this.componentManager.getInstanceList(HibernateDataMigration.class);
+            Set<String> ignoredMigrations = new HashSet<>(this.hibernateConfiguration.getIgnoredMigrations());
+            Map<XWikiDBVersion, String> hintMap = new HashMap<>();
+            for (HibernateDataMigration dataMigration : migrationList) {
+                XWikiDBVersion version = dataMigration.getVersion();
+                if (hintMap.containsKey(version)) {
+                    if (isMigrationIgnored(dataMigration, ignoredMigrations)) {
```

Comment:
What if the ignored migration is processed second?
iiuc the first ignored migration is going to be added to `hintMap`, and the second migration will trigger a `InitializationException` instead of a warn log.
Is my understanding correct?

Reply:
Yes and for me it's ok: the only reason why I make exception here for ignored migration is to not block users: i.e. they can use the ignored migration feature to skip this check if they need to. Ignored migration can use either the version of the migration, or the name. In theory this should never be used, but I imagine weird UC where there's a migration coming from an extension which have same DB version: we probably don't want to block the initialization of the wiki.

Reply:
ok, make sense. Thanks
Might be worth having a comment, to avoid confusion when reading this code later.